File: backend/app/schemas/calendar.py
from pydantic import BaseModel
from typing import Optional, Literal
from datetime import datetime
from uuid import UUID

class CalendarEvent(BaseModel):
    id: UUID
    title: str
    type: Literal["TASK", "SESSION", "ASSESSMENT", "REVIEW"]
    date: datetime
    end_date: Optional[datetime] = None
    status: Optional[str] = None
    subject_name: Optional[str] = None
    color: Optional[str] = None

    class Config:
        from_attributes = True

File: frontend/src/types/calendar.ts
export interface CalendarEvent {
  id: string;
  title: string;
  type: "TASK" | "SESSION" | "ASSESSMENT";
  date: string;
  end_date?: string | null;
  status?: string | null;
  subject_name?: string | null;
  color?: string | null;
}
